lms/djangoapps/program_enrollments/api/v1/tests/test_views.py

```diff
@@ -81,35 +81,130 @@ def get_url(self, program_uuid=None, course_id=None):
-    view_name = 'programs_api:v1:learner_program_enrollments'
+    view_name = 'programs_api:v1:user_program_readonly_access'
+
+    @classmethod
+    def setUpClass(cls):
+        super(UserProgramReadOnlyAccessViewTest, cls).setUpClass()
+
+        cls.many_programs_return_list = [
+            {'uuid': cls.program_uuid_tmpl.format(11), 'marketing_slug': 'garbage-program', 'type': 'masters'},
+            {'uuid': cls.program_uuid_tmpl.format(22), 'marketing_slug': 'garbage-study', 'type': 'micromaster'},
+            {'uuid': cls.program_uuid_tmpl.format(33), 'marketing_slug': 'garbage-life', 'type': 'masters'},
+        ]
+
+        cls.course_staff = InstructorFactory.create(password=cls.password, course_key=cls.course_id)
+        cls.date = datetime(2013, 1, 22, tzinfo=UTC)
+        CourseEnrollmentFactory(
+            course_id=cls.course_id,
+            user=cls.course_staff,
+            created=cls.date,
+        )
 
     def test_401_if_anonymous(self):
         response = self.client.get(reverse(self.view_name))
         assert status.HTTP_401_UNAUTHORIZED == response.status_code
 
+    def _assert_response_and_mock(self, response, mock_get_programs, call_list, data_size):
+        assert status.HTTP_200_OK == response.status_code
+        assert len(response.data) == data_size
+        mock_get_programs.assert_has_calls(call_list)
+
+    @ddt.data(
+        ('masters', 2),
+        ('micromaster', 1)
+    )
+    @ddt.unpack
+    def test_global_staff(self, program_type, data_size):
+        self.client.login(username=self.global_staff.username, password=self.password)
+        with mock.patch(
+            'lms.djangoapps.program_enrollments.api.v1.views.get_programs',
+            autospec=True,
+            return_value=self.many_programs_return_list
+        ) as mock_get_programs:
+            response = self.client.get(reverse(self.view_name) + '?type=' + program_type)
+            call_list = [mock.call(response.wsgi_request.site)]
+            self._assert_response_and_mock(response, mock_get_programs, call_list, data_size)
+
+    def test_course_staff(self):
+        self.client.login(
+            username=self.course_staff.username,
+            password=self.password
+        )
+        with mock.patch(
+            'lms.djangoapps.program_enrollments.api.v1.views.get_programs',
+            autospec=True,
+            return_value=[{'uuid': 'boop', 'marketing_slug': 'garbage-program', 'type': 'masters'}]
+        ) as mock_get_programs:
+            response = self.client.get(reverse(self.view_name) + '?type=masters')
+            call_list = [mock.call(course=self.course_id)]
+            self._assert_response_and_mock(response, mock_get_programs, call_list, 1)
+
+    def test_course_staff_of_multiple_courses(self):
+        other_course_key_string = 'course-v1:edX+ToyX+Other_Course'
+        other_course_key = CourseKey.from_string(other_course_key_string)
+
+        # add a course over view for other_course_key
```

Comment:
nit: "overview" should be one word.

Reply:
although, another nit: I question the value of this comment - it's pretty clear what the code does.